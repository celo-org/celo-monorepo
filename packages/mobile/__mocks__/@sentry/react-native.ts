const Sentry = {
  config: () => ({ install: jest.fn() }),
  setTagsContext: jest.fn(),
  captureException: jest.fn(),
}

module.exports = Sentry
