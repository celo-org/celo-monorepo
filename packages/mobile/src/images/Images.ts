export const welcomeRocketIcon = require('src/images/welcome-rocket-icon.png')
export const backupIcon = require('src/images/backup-icon.png')
export const clockIcon = require('src/images/clock-icon.png')
export const homeIcon = require('src/images/home-icon.png')
export const profileIcon = require('src/images/profile-icon.png')
export const settingsIcon = require('src/images/settings-icon.png')
export const getVerifiedIcon = require('src/images/get-verified-icon.png')
export const stableCurrencyEducation = require('src/images/stable-currency-education.png')
export const placeholderElipse = require('src/images/placeholder-elipse.png')
export const welcomeIcons = require('src/images/welcome-icons.png')
export const placeholder = require('src/images/placeholder.png')
export const exchangeGold = require('src/images/exchangeGold.png')
export const shinyDollar = require('src/images/shiny-dollar.png')
export const shinyGold = require('src/images/shiny-gold.png')
export const threeChecks = require('src/images/3-checks.png')
export const addressBook = require('src/images/address-book.png')
export const bigPhoneAvatar = require('src/images/big-phone-avatar.png')
export const bigStability = require('src/images/big-stability.png')
export const cameraUpload = require('src/images/camera-upload.png')
export const exchangeIcon = require('src/images/exchange.png')
export const findUser = require('src/images/find-user.png')
export const goldValue = require('src/images/gold-value.png')
export const sendBetweenPhones = require('src/images/send-between-phones.png')
export const sendFee = require('src/images/send-fee.png')
export const sendReceive = require('src/images/send-receive.png')
export const stabilityScale = require('src/images/stability-scale.png')
export const verifyAddressBook = require('src/images/verify-address-book.png')
export const verifyPhone = require('src/images/verify-phone.png')
export const faucetIcon = require('src/images/coins-logo.png')
export const coinsIcon = require('src/images/coins-logo.png')
export const rewardsAppIcon = require('src/images/Rewards-app-icon.png')
export const inviteFriendsIcon = require('src/images/Invite-Friends.png')
export const sendDollar = require('src/images/send-dollar.png')
export const unknownUserIcon = require('src/images/unknown-user-icon.png')
export const ringsPlusCircle = require('src/images/rings-plus-circle.png')
export const contactPermission = require('src/images/contact-permission.png')
// it looks like jests tests fail when a sub component references image via this file rather than require direct
