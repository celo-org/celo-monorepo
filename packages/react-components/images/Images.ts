export const unknownUserIcon = require('@celo/react-components/images/unknown-user-icon.png')
