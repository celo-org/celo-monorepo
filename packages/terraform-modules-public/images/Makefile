alfajores-full:
	packer build \
		-var 'sync_mode=full' \
		-var 'additional_params="--alfajores"' \
		-var 'network_name=alfajores' \
		-force node.json

alfajores-lightest:
	packer build \
		-var 'sync_mode=lightest' \
		-var 'additional_params="--alfajores"' \
		-var 'network_name=alfajores' \
		-force node.json

mainnet-full:
	packer build \
		-var 'sync_mode=full' \
		-var 'network_name=mainnet' \
		-force node.json

mainnet-lightest:
	packer build \
		-var 'sync_mode=lightest' \
		-var 'network_name=mainnet' \
		-force node.json

alfajores: alfajores-lightest alfajores-full

mainnet: mainnet-lightest mainnet-full

.PHONY: alfajores alfajores-full alfajores-lightest mainnet mainnet-full mainnet-lightest