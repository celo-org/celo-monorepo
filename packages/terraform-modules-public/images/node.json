{
  "variables": {
    "aws_access_key": "{{env `AWS_ACCESS_KEY_ID`}}",
    "aws_secret_key": "{{env `AWS_SECRET_ACCESS_KEY`}}",
    "azure_client_id": "{{env `AZURE_CLIENT_ID`}}",
    "azure_client_secret": "{{env `AZURE_CLIENT_SECRET`}}",
    "azure_tenant_id": "{{env `AZURE_TENANT_ID`}}",
    "azure_subscription_id": "{{env `AZURE_SUBSCRIPTION_ID`}}",
    "gcp_project_id": "{{env `GCP_PROJECT_ID`}}",
    "region": "us-east-1",
    "sync_mode": "full",
    "network_name": "mainnet",
    "additional_params": ""
  },
  "builders": [
    {
      "name": "aws",
      "type": "amazon-ebs",
      "access_key": "{{user `aws_access_key`}}",
      "secret_key": "{{user `aws_secret_key`}}",
      "ami_name": "celo-{{user `network_name`}}-{{user `sync_mode`}}-node-latest",
      "instance_type": "t2.medium",
      "region": "{{user `region`}}",
      "source_ami_filter": {
        "filters": {
          "virtualization-type": "hvm",
          "name": "ubuntu/images/*ubuntu-bionic-18.04-amd64-server-*",
          "root-device-type": "ebs"
        },
        "owners": ["099720109477"],
        "most_recent": true
      },
      "ssh_username": "ubuntu"
    },
    {
      "name": "azure",
      "type": "azure-arm",
      "client_id": "{{user `azure_client_id`}}",
      "client_secret": "{{user `azure_client_secret`}}",
      "tenant_id": "{{user `azure_tenant_id`}}",
      "subscription_id": "{{user `azure_subscription_id`}}",
      "image_publisher": "Canonical",
      "image_offer": "UbuntuServer",
      "image_sku": "18.04-LTS",
      "managed_image_name": "celo-{{user `network_name`}}-{{user `sync_mode`}}-node-latest",
      "managed_image_resource_group_name": "CeloNodeImages",
      "location": "West Us 2",
      "temp_resource_group_name": "CeloNodeImageBuilder",
      "os_type": "Linux",
      "ssh_username": "ubuntu"
    },
    {
      "name": "gcp",
      "type": "googlecompute",
      "project_id": "{{user `gcp_project_id`}}",
      "source_image": "ubuntu-1804-bionic-v20200317",
      "image_name": "celo-{{user `network_name`}}-{{user `sync_mode`}}-node-latest",
      "ssh_username": "ubuntu",
      "zone": "us-central1-a"
    }
  ],
  "provisioners": [
    {
      "type": "shell",
      "execute_command": "echo 'packer' | sudo -S sh -c '{{ .Vars }} {{ .Path }}'",
      "script": "../aws/testnet/modules/startup-scripts/install-base.sh"
    },
    {
      "type": "shell",
      "execute_command": "echo 'packer' | sudo -S sh -c '{{ .Vars }} {{ .Path }}'",
      "script": "../aws/testnet/modules/startup-scripts/install-docker.sh"
    },
    {
      "type": "shell",
      "inline": [
        "echo {{user `additional_params`}} >> additional_params",
        "echo {{user `sync_mode`}} >> sync_mode"
      ]
    },
    {
      "type": "shell",
      "execute_command": "echo 'packer' | sudo -S sh -c '{{ .Vars }} {{ .Path }}'",
      "script": "scripts/init.sh"
    },
    {
      "type": "file",
      "source": "scripts/start.sh",
      "destination": "/tmp/start.sh"
    },
    {
      "type": "shell",
      "execute_command": "echo 'packer' | sudo -S sh -c '{{ .Vars }} {{ .Path }}'",
      "inline": [
        "cp /tmp/start.sh /var/lib/cloud/scripts/per-instance/start.sh",
        "chmod +x /var/lib/cloud/scripts/per-instance/start.sh"
      ]
    },
    {
      "type": "shell",
      "execute_command": "echo 'packer' | sudo -S sh -c '{{ .Vars }} {{ .Path }}'",
      "inline": ["find / -name \"authorized_keys\" -exec rm -f {} \\;"]
    }
  ]
}
