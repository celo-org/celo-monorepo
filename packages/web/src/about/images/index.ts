import BeautifulMoneyPreview from './BeautifulMoneyPreview.jpg'
import mintPlaza from './mint-plaza@2x.png'
import teamHero from './team-hero@2x.png'

export { BeautifulMoneyPreview, mintPlaza, teamHero }
