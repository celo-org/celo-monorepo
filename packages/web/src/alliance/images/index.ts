import aguaComunidad from 'src/alliance/images/agua-comunidad.jpg'
import allianceOG from 'src/alliance/images/Alliance-OG.jpg'
import clabsMobile from 'src/alliance/images/clabs-mobile.jpg'
import clabsPreview from 'src/alliance/images/clabs-preview.jpg'
import clabs from 'src/alliance/images/clabs.jpg'
import fences from 'src/alliance/images/fences.jpg'
import laboratoriaMobile from 'src/alliance/images/laboratoria-mobile.jpg'
import laboratoriaPreview from 'src/alliance/images/laboratoria-preview.jpg'
import laboratoria from 'src/alliance/images/laboratoria.jpg'
import wfpMobile from 'src/alliance/images/wfp-mobile.jpg'
import wfpPreview from 'src/alliance/images/wfp-preview.jpg'
import wfp from 'src/alliance/images/wfp.jpg'

export {
  fences,
  clabs,
  clabsPreview,
  clabsMobile,
  laboratoria,
  laboratoriaMobile,
  laboratoriaPreview,
  wfp,
  wfpMobile,
  wfpPreview,
  allianceOG,
  aguaComunidad,
}
