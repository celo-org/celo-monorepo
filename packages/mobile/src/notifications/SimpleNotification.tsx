import BaseNotification, { CTA } from '@celo/react-components/components/BaseNotification'
import fontStyles from '@celo/react-components/styles/fonts'
import * as React from 'react'
import { Image, ImageSourcePropType, StyleSheet, Text, View } from 'react-native'

interface OwnProps {
  text: string
  title: string
  image: ImageSourcePropType
  ctaList: CTA[]
}

type Props = OwnProps

export function SimpleNotification(props: Props) {
  return (
    <BaseNotification
      title={props.title}
      icon={<Image source={props.image} style={styles.image} resizeMode="contain" />}
      ctas={props.ctaList}
    >
      <View style={styles.body}>
        <Text style={fontStyles.subSmall}>{props.text}</Text>
      </View>
    </BaseNotification>
  )
}

const styles = StyleSheet.create({
  body: {
    marginTop: 5,
  },
  image: {
    width: 30,
    height: 30,
  },
})

export default SimpleNotification
