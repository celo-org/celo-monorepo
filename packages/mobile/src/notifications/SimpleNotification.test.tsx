import * as React from 'react'
import 'react-native'
import * as renderer from 'react-test-renderer'
import { placeholder } from 'src/images/Images'
import SimpleNotification from 'src/notifications/SimpleNotification'

const props = () => ({
  text: 'Gold is where you can choose to store Celo dollars you have',
  image: placeholder,
  title: 'Test',
  ctaList: [{ text: 'it goes boom', onPress: jest.fn() }],
})

describe(SimpleNotification, () => {
  it('renders correctly', () => {
    const tree = renderer.create(<SimpleNotification {...props()} />)
    expect(tree).toMatchSnapshot()
  })
})
