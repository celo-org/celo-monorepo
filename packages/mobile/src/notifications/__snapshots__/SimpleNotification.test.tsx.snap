// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleNotification renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 16,
        "width": "100%",
      },
      Object {
        "borderColor": "#EEEEEE",
        "borderRadius": 3,
        "borderWidth": 1,
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "paddingRight": 16,
      }
    }
  >
    <Image
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../packages/mobile/src/images/placeholder.png",
        }
      }
      style={
        Object {
          "height": 30,
          "width": 30,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#2E3338",
          "fontFamily": "Hind-Bold",
          "fontSize": 14,
          "lineHeight": 18,
        }
      }
    >
      Test
    </Text>
    <View>
      <View
        style={
          Object {
            "justifyContent": "space-between",
            "minHeight": 70,
          }
        }
      >
        <View
          style={
            Object {
              "marginTop": 5,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#81868B",
                "fontFamily": "Hind-Regular",
                "fontSize": 14,
                "lineHeight": 20,
              }
            }
          >
            Gold is where you can choose to store Celo dollars you have
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "marginTop": 15,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            nativeBackgroundAndroid={
              Object {
                "attribute": "selectableItemBackgroundBorderless",
                "type": "ThemeAttrAndroid",
              }
            }
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#42D689",
                    "fontFamily": "Hind-SemiBold",
                    "fontSize": 14,
                    "lineHeight": 18,
                  },
                  Object {
                    "paddingEnd": 15,
                  },
                ]
              }
            >
              it goes boom
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
